File: src/main/java/com/example/sns/dto/request/PostUpdateRequest.java
package com.example.sns.dto.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

/**
 * 게시글 수정 요청.
 * 프론트엔드 호환: imageUrl, category 필드 추가.
 */
public record PostUpdateRequest(
        @NotBlank(message = "제목은 필수입니다.")
        @Size(max = 200)
        String title,

        @NotBlank(message = "내용은 필수입니다.")
        String content,

        Double latitude,
        Double longitude,
        Long pinId,
        String imageUrl,
        String category
) {
}

File: mobile/lib/domain/models/post_update_request.dart
/// doc/API_SPEC.md 4.4 — PostUpdateRequest
class PostUpdateRequest {
  const PostUpdateRequest({
    this.title,
    this.content,
    this.latitude,
    this.longitude,
    this.pinId,
  });

  final String? title;
  final String? content;
  final double? latitude;
  final double? longitude;
  final int? pinId;

  Map<String, dynamic> toJson() => {
        if (title != null) 'title': title,
        if (content != null) 'content': content,
        if (latitude != null) 'latitude': latitude,
        if (longitude != null) 'longitude': longitude,
        if (pinId != null) 'pinId': pinId,
      };
}
